From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 25 Aug 2026 19:47:48 +0200
Subject: fix(convertfs): support find applet from busybox

busybox's `find` applet does not support `-delete`. So replace `-delete`
by a `rm` call.

Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/deb14ebbfed325939c15289e92b4a8836167df32
---
 modules.d/30convertfs/convertfs.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/30convertfs/convertfs.sh b/modules.d/30convertfs/convertfs.sh
index 0b3b00d..26a2f12 100755
--- a/modules.d/30convertfs/convertfs.sh
+++ b/modules.d/30convertfs/convertfs.sh
@@ -109,7 +109,7 @@ for dir in bin sbin lib lib64; do
     cp -axT ${CP_HARDLINK:+"$CP_HARDLINK"} --backup --suffix=.usrmove~ "$ROOT/$dir" "$ROOT/usr/${dir}.usrmove-new"
     echo "Clean up duplicates in \`$ROOT/usr/$dir'."
     # delete all symlinks that have been backed up
-    find "$ROOT/usr/${dir}.usrmove-new" -type l -name '*.usrmove~' -delete || :
+    find "$ROOT/usr/${dir}.usrmove-new" -type l -name '*.usrmove~' -exec rm -f {} + || :
     # replace symlink with backed up binary
     # shellcheck disable=SC2156
     find "$ROOT/usr/${dir}.usrmove-new" \
